container.go

```diff
@@ -50,7 +50,7 @@ type Container interface {
 	Stop(context.Context, *time.Duration) error                                    // stop the container
 
 	// Terminate stops and removes the container and its image if it was built and not flagged as kept.
-	Terminate(ctx context.Context) error
+	Terminate(ctx context.Context, opts ...TerminateOption) error
```

Comment:
Hm, I wonder why we need a separate parameter for the timeout when a context is already provided to `Terminate()`?

As a user, I expect this to work fine. Does this code not do what is expected?

```go
ctx, _ := context.WithTimeout(context.Background(), time.Second*10)
c.Terminate()
```

If a `Duration` value is needed (vs reacting to a channel close), `context.Background().Deadline()` returns a deadline `Time`. It seems that there is no need for a new API.